server/WAPI.rb

```diff
@@ -66,7 +66,7 @@ def initialize(application)
     @uniqname = application['uniqname']
 
     @renewal = WAPI.build_renewal(@key, @secret)
-    logger.info("WAPI: #{__LINE__}: initialize WAPI with #{@api_prefix}")
+    logger.info("#{self.class.to_s}:#{__method__}:#{__LINE__}: initialize WAPI with #{@api_prefix}")
```

Comment:
Great idea but I don't know how to do that and get the __method__ and
__LINE__ special variables to have the right values.

On Fri, Nov 6, 2015 at 10:09 AM, Zhen Qian <notifications@github.com> wrote:

> In server/WAPI.rb
> <https://github.com/tl-its-umich-edu/StudentDashboard/pull/213#discussion_r44145653>
> :
>
> > @@ -66,7 +66,7 @@ def initialize(application)
> >      @uniqname = application['uniqname']
> >
> >      @renewal = WAPI.build_renewal(@key, @secret)
> > -    logger.info("WAPI: #{__LINE__}: initialize WAPI with #{@api_prefix}")
> > +    logger.info("#{self.class.to_s}:#{__method__}:#{__LINE__}: initialize WAPI with #{@api_prefix}")
>
> I wonder whether it is possible to make a util routine for log String
> format: "#{self.class.to_s}:#{*method*}:#{*LINE*}"
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tl-its-umich-edu/StudentDashboard/pull/213/files#r44145653>
> .
>